### DISCUSSION TITLE
Testing RESTful APIs

### DISCUSSION THREAD

**Discussion by alberbm-dev** [DHI/python-package-development#14] [Q&A]
Hello everyone! :wave: 

I really enjoyed yesterday's session about testing in Python, as I think it's a crucial skill to develop quality code. I, however, would have liked to learn more about testing more complex elements, such as RESTful APIs. I have created some tests for these myself in the past, but I still find myself wondering how to improve them. I usually have questions regarding mocking endpoints and their responses, which tests and assertions make sense to include, etc.

Does anybody have good insights or knows any quality source of information about this topic?

**Comment by jsmariegaard** [comment]
Testing of REST APIs is outside the scope of this course, but surely an interesting topic. Personally, not something I have a lot of experience but a colleague suggested this: 

* https://fastapi.tiangolo.com/tutorial/testing/ 
* https://www.lambdatest.com/learning-hub/pytest-api-testing 

Does it help?

**Reply by alberbm-dev** [reply]
It does! Thank you very much :D
